### ISSUE TITLE
chapter 7 integrated model does not converge

### ISSUE THREAD

**Issue by ChrisFishCahill** [James-Thorson/Spatio-temporal-models-for-ecologists#2]
The `integrated_model.R` example from Chapter 7 nlminb returns a convergence code of 1, indicating non-convergence.  The issue can be replicated by running the book script through line 104 and then inspecting the output of `opt$convergence`. 

This issue is somewhat suppressed for users as the book code then places `opt$SD <- sdreport(obj)` in [line 105.](https://github.com/James-Thorson/Spatio-temporal-models-for-ecologists/blob/main/Chap_7/integrated_model.R#L105C1-L105C49)

**Comment by kaskr** [comment]
The problem seems to be that this is a close to singular fit, and results may differ across OS/cpu etc. I get a seemingly valid fit:

```r
> opt$SD
sdreport(.) result
               Estimate  Std. Error
ln_tau       -3.1448016  0.17877420
ln_kappa      0.6744774  0.23758758
ln_phi        3.4518348  0.02403412
finv_power    0.4987681  0.02445525
gamma_k     -71.6944986  7.62542623
gamma_k    -722.8524069 49.91189962
gamma_k       0.4006631  0.82832450
gamma_k       0.7038825  0.82873979
gamma_k       0.5274318  0.82764320
gamma_k       0.7354822  0.82898211
gamma_k       0.2217676  0.82757737
gamma_k       0.6791444  0.82693115
gamma_k       1.6211728  0.82891350
gamma_k       1.1414576  0.82864313
gamma_k       0.2955675  0.82934260
eta_j        -2.8170148  0.04990204
eta_j        -6.9520215  0.08411494
Maximum gradient component: 0.004966166 
```

 with

```r
> opt$SD$pdHess
[1] TRUE
```

The smallest hessian eigen value is  ca `1e-4` and points in direction of `gamma_k[2]` (not surprising).

**Comment by kaskr** [comment]
@ChrisFishCahill Can you try to rescale the the first two covariates:

```r
X_ik[,1:2] <- X_ik[,1:2]*100
```

That improves the conditioning a lot for me (ratio of largest vs smallest eigenvalue went from `13829725` to `33587.08`) and the optimization problem runs smoothly.

**Comment by ChrisFishCahill** [comment]
@kaskr I just tested this fix and the optimization now runs smoothly for me. Thanks!

**Comment by James-Thorson-NOAA** [comment]
Do either of you have any suggestions about how I can track fixes for typos? 

I was thinking I could make a NEWS doc that tracks typo-fixes, and also do a new release 1.0.1 at some point that updates the code and LaTeX files ... any better idea for how to address typos and small fixes like this?

**Comment by ChrisFishCahill** [comment]
I am not sure of the best solution here @James-Thorson-NOAA. However, I was going to ask something along those lines and whether you wanted each small issue posted on the issues page as their own issues or whether there is something more efficient for you.  I have a collection of small things (e.g., like R scripts erroring out before running TMB on some scripts on my machine) that I've found during my conversion of the book examples to RTMB for teaching purposes that might be of use to you.
